src/main/java/org/skriptlang/skript/lang/experiment/ExperimentData.java

```diff
@@ -0,0 +1,219 @@
+package org.skriptlang.skript.lang.experiment;
+
+import ch.njol.skript.Skript;
+import ch.njol.util.StringUtils;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnmodifiableView;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Container for holding {@link Experiment}s that must be enabled or disabled to use.
+ */
+public class ExperimentData {
+
+	/**
+	 * Create and return a new {@link Builder}.
+	 */
+	public static Builder builder() {
+		return new Builder();
+	}
+
+	/**
+	 * Create a new {@link ExperimentData} with only a singular required {@link Experiment}
+	 * @param experiment The required {@link Experiment}.
+	 * @return {@link ExperimentData}
+	 */
+	public static ExperimentData createSingularData(Experiment experiment) {
+		return builder().required(experiment).build();
+	}
+
+	private final Set<Experiment> required;
+	private final Set<Experiment> disallowed;
+	private final String errorMessage;
+
+	/**
+	 * @param required {@link Set} of {@link Experiment}s required to be enabled.
+	 * @param disallowed {@link Set} of {@link Experiment}s required to be disabled.
+	 * @param errorMessage {@link String} used to error in {@link #checkRequirementsAndError(ExperimentSet)}
+	 *                     If {@code null}, will use {@link #constructError()}.
+	 */
+	private ExperimentData(Set<Experiment> required, Set<Experiment> disallowed, @Nullable String errorMessage) {
+		this.required = required;
+		this.disallowed = disallowed;
+		this.errorMessage = errorMessage != null ? errorMessage : constructError();
+	}
+
+	/**
+	 * Get the {@link Experiment}s that must be enabled in order to use.
+	 */
+	public @UnmodifiableView Set<Experiment> getRequired() {
+		return Collections.unmodifiableSet(required);
```

Comment:
You could always store them as unmodifiable sets. Also, I'm changing my mind on UnmodifiableView, as that implies the underlying collection could change (which should not be the case). You can revert to Unmodifiable